### ISSUE TITLE
Invalid mock generation for functions with typealiased return types

### ISSUE THREAD

**Issue by xpathexception** [kosi-libs/MocKMP#34]
``` kotlin
typealias SampleIntKeyMap<T> = Map<Int, T>

interface Sample {
    fun sample(): SampleIntKeyMap<String>
}
```
Generated mock implemenation for this sample has invalid type `T` as `Map#value` type:
``` kotlin
import org.kodein.mock.Mocker

internal class MockSample(
    private val mocker: Mocker,
) : Sample {
    public override fun sample(): Map<Int, T> = this.mocker.register( //<=== Map<Int, String> or SampleIntKeyMap<String> is expected
        receiver = this,
        method = "sample()"
    )

    public override fun toString(): String = this.mocker.register(
        receiver = this,
        method = "toString()",
        default = { super.toString() }
    )
}

```

**Comment by ankushg** [comment]
@SalomonBrys looks like the issue was closed in KSP!

**Comment by afonsograca** [comment]
Hi @SalomonBrys I believe Kotlin Poet has just released 1.13.0. Would that version work?
